tests/test_bgp_neighbor.rb

```diff
@@ -23,89 +23,132 @@
 
 # TestRouterBgpNeighbor - Minitest for RouterBgpNeighbor node utility class
 class TestRouterBgpNeighbor < CiscoTestCase
-  # rubocop:disable Style/ClassVars
-  @@asn = 55
-  @@addr = '1.1.1.1'
-  # rubocop:enable Style/ClassVars
+  ASN = 55
+  ADDR = '1.1.1.1'
+  REMOTE_ASN = 99
 
   def setup
     # Disable feature bgp before each test to ensure we
     # are starting with a clean slate for each test.
     super
-    config('no feature bgp', 'feature bgp', 'router bgp 55')
+    if platform == :ios_xr
+      config('no router bgp', 'router bgp 55')
+    else
+      config('no feature bgp', 'feature bgp', 'router bgp 55')
+    end
   end
 
   def teardown
-    config('no feature bgp')
+    disable_bgp
   end
 
-  def get_bgpneighbor_match_line(addr, vrf='default')
-    if vrf == 'default'
-      cmd = "show run bgp all | section 'router bgp' | no-more"
+  def disable_bgp
+    if platform == :ios_xr
+      config('no router bgp')
     else
-      cmd = "show run bgp all | section 'vrf #{vrf}' | no-more"
+      config('no feature bgp')
     end
-    CiscoLogger.debug("matching #{addr} with vrf #{vrf}")
-    s = @device.cmd("#{cmd}")
-    line = /neighbor #{addr}/.match(s)
-    CiscoLogger.debug(line)
-    line
   end
 
-  def test_bgpneighbor_collection_empty
-    config('no feature bgp')
+  # Returns some test data for use in several tests.
+  def test_data
+    test_data = []
+    test_data << { vrf: 'default', neighbors: ['1.1.1.1'] }
+
+    if platform == :ios_xr
+      # XR doesn't support prefix/len addresses
+      test_data << { vrf:       'red',
+                     neighbors: ['2.2.2.0', '2000::2', '2000:123:38::'] }
```

Comment:
Given that XR doesn't support prefix/len addresses, I wonder if we would be faster (and just as thorough in our XR coverage) if we removed the extra addresses instead of just stripping the mask:

`neighbors: ['2.2.2.0', '2000::2']`